src/components/Navbar/components/MenuNavbar/components/MenuIcon.tsx

```diff
@@ -0,0 +1,72 @@
+import { Badge, Box, Stack, Typography } from '@mui/material'
+import { styled } from '@mui/material/styles'
+import { FC, ReactNode } from 'react'
+
+// -------------------- styled components --------------------
+
+const IconWrapper = styled(Stack)(({ theme }) => ({
+  position: 'relative',
+  width: '24px',
+  height: '24px',
+  cursor: 'pointer',
+  borderRadius: '100%',
+
+  '&:hover': {
+    '& .icon-hover': {
+      backgroundColor: theme.palette.neutral700.main
+    }
+  },
+
+  '& .MuiSvgIcon-root': {
+    width: '20px',
+    height: '20px',
+    color: theme.palette.neutral500.main
+  }
+}))
+
+const BgHover = styled(Box)(() => ({
+  width: '32px',
+  height: '32px',
+  borderRadius: '100%',
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  transition: 'background-color 0.3s'
+}))
+
+// -------------------------- types --------------------------
+
+type MenuIconProps = {
+  value: number
+  onClick?: () => void
+  children: ReactNode
+}
+
+// ----------------------- components ------------------------
+
+const MenuIcon: FC<MenuIconProps> = ({ value, onClick, children }) => {
```

Comment:
```suggestion
const MenuIcon: FC<MenuIconProps> = ({ value, onClick, children }) => {
```
props disini ada children kan ya? khusus buat children bisa pake type ini
`React.PropsWithChildren`
jadi nanti bisa bikin nya cukup
```
interface Props extends React.PropsWithChildren {
   name: string
   priority: "bold" | "normal"
}
```
props.children otomatis ada harusnya